packages/main/src/services/trayMenu/index.ts

```diff
@@ -1,40 +1,149 @@
 import { injectable, inject } from 'inversify';
 import { Menu, app, Tray, nativeImage } from 'electron';
+import { NuclearMeta, IpcEvents} from '@nuclear/core';
 import HttpApi from '../http';
 import Config from '../config';
 import Platform from '../platform';
+import Window from '../window';
+
+
+type PlayerContext = {
+  isPlaying?: boolean;
+  track?: NuclearMeta
+}
 
 @injectable()
 class TrayMenu {
-  private menu: Menu;
   private tray: Tray;
 
+  private playerContext: PlayerContext;
+
   constructor(
     @inject(Config) private config: Config,
     @inject(HttpApi) private httpApi: HttpApi,
-    @inject(Platform) private platform: Platform
-  ) {}
+    @inject(Platform) private platform: Platform,
+    @inject(Window) private window: Window
+  ) { }
 
   init() {
-    this.menu = Menu.buildFromTemplate([
-      {
-        label: 'Quit',
+    const icon = nativeImage.createFromPath(
+      this.platform.isMac() ? this.config.macIcon : this.config.icon
+    );
+    this.tray = new Tray(icon);
+
+    if (!this.platform.isMac()) {
+      this.tray.setTitle(this.config.title);
+    }
+
+    this.setPlayerContext({
+      isPlaying: false,
+      track: null
+    });
+    this.tray.setToolTip(this.getToolTipString());
+    this.tray.setContextMenu(this.getMenu());
+  }
+
+  getMenu() {
+    const template = [];
+
+    // Playing status
+    if (this.playerContext.track) {
+      template.push({
+        label: `${this.playerContext.isPlaying ? 'Playing - ' : ''}${this.playerContext.track.name}`,
+        enabled: false
+      });
+      template.push({
+        label: `by ${this.playerContext.track.artist}`,
+        enabled: false
+      });
+    } else {
+      template.push({
+        label: '--/--',
+        enabled: false
+      });
+    }
+    
+    template.push({
+      type: 'separator'
+    });
+
+    // Control button
+    if (this.playerContext.track) {
+      if (this.playerContext.isPlaying) {
+        template.push({
+          label: 'Pause',
+          type: 'normal',
+          click: async () => {
+            this.window.send(IpcEvents.PAUSE);
+            this.update({isPlaying: false});
+          }
+        });
+      } else {
+        template.push({
+          label: 'Play',
+          type: 'normal',
+          click: async () => {
+            this.window.send(IpcEvents.PLAY);
+            this.update({isPlaying: true});
+          }
+        });
+      }
+
+      template.push({
+        label: 'Next',
         type: 'normal',
         click: async () => {
-          await this.httpApi.close();
+          this.window.send(IpcEvents.NEXT);
+        }
+      });
 
-          app.quit();
+      template.push({
+        label: 'Previous',
+        type: 'normal',
+        click: async () => {
+          this.window.send(IpcEvents.PREVIOUS);
```

Comment:
This one doesn't work as expected, it jumps from the top track to the bottom

Reply:
If you're at the first track, clicking previous should loop to the last track.